QModManager/API/ModLoading/Internal/Initializer.cs

```diff
@@ -0,0 +1,178 @@
+﻿namespace QModManager.API.ModLoading
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using Harmony;
+    using QModManager.API.ModLoading.Internal;
+    using QModManager.API.SMLHelper.Patchers;
+    using QModManager.Utility;
+
+    internal class Initializer
+    {
+        private readonly Game currentGame;
+        internal readonly IDictionary<string, ModLoadingResults> NonLoadedMods = new Dictionary<string, ModLoadingResults>();
+        internal readonly IDictionary<ModLoadingResults, int> ErrorTotals = new Dictionary<ModLoadingResults, int>
+        {
+            { ModLoadingResults.Failure, 0 },
+            { ModLoadingResults.AlreadyLoaded, 0 },
+            { ModLoadingResults.CurrentGameNotSupported, 0 },
+        };
+
+        internal int FailedToLoad { get; private set; }
+
+        internal Initializer(Game currentlyRunningGame)
+        {
+            currentGame = currentlyRunningGame;
+        }
+
+        internal void InitializeMods<Q>(ICollection<Q> modsToInitialize)
+            where Q : IQModLoadable
+        {
+            InitializeMods(modsToInitialize, PatchingOrder.PreInitialize);
+            InitializeMods(modsToInitialize, PatchingOrder.NormalInitialize);
+            InitializeMods(modsToInitialize, PatchingOrder.PostInitialize);
+            FinalInitialize();
+
+            LogResults(ModLoadingResults.Failure, "The following mods failed to load to the errors during their initialization");
+            LogResults(ModLoadingResults.AlreadyLoaded, "The following mods encountered duplicate initialization attempts");
+            LogResults(ModLoadingResults.CurrentGameNotSupported, $"The following mods for '{GetOtherGame()}' were skipped");
```

Comment:
Should this logging be left to another class?

Reply:
So far, it seems like each class is doing its own logging, but logging might be something we can centralize, leaving these classes to be purely functional.  
I'll be looking into it later.